metrics-scraper/monitoring.go

```diff
@@ -91,66 +125,84 @@ func run(
 		case <-ticker.C:
 			if err := pollData(ctx, c, config.Moniker, networkID); err != nil {
 				log.Printf("error polling data for %v: %v", config.Moniker, err)
+			} else {
+				log.Printf("data polled successfully for: %v", config.Moniker)
 			}
 		}
 	}
 }
 
 func pollData(ctx context.Context, client *rpc.Client, moniker, networkID string) error {
+	errs := []error{}
 	if err := sidecarsVersion(ctx, client, moniker, networkID); err != nil {
-		return err
+		errs = append(errs, err)
 	}
 
 	if err := nodeVersion(ctx, client, moniker, networkID); err != nil {
-		return err
+		errs = append(errs, err)
 	}
 
 	if err := latestBlockAndTimestamp(ctx, client, moniker, networkID); err != nil {
-		return err
+		errs = append(errs, err)
 	}
 
-	return nil
+	return errors.Join(errs...)
 }
 
 func latestBlockAndTimestamp(ctx context.Context, client *rpc.Client, moniker, networkID string) error {
 	result := struct {
 		Number    string `json:"number"`
 		Timestamp string `json:"timestamp"`
 	}{}
+
 	err := client.CallContext(ctx, &result, ethGetBlockByNumberEndpoint, "latest", false)
 	if err != nil {
 		mezodUpGauge.WithLabelValues(moniker, networkID).Set(0)
+		// set latest block to 0 for error
+		mezoLatestBlockGauge.WithLabelValues(moniker, networkID).Set(0)
+		// set latest timestamp to 0 for error
+		mezoLatestTimestampGauge.WithLabelValues(moniker, networkID).Set(0)
 		return fmt.Errorf("couldn't call %v for %v: %v", ethGetBlockByNumberEndpoint, moniker, err)
 	}
 
 	mezodUpGauge.WithLabelValues(moniker, networkID).Set(1)
 
+	errs := []error{}
 	latestBlock, err := strconv.ParseUint(strings.TrimPrefix(result.Number, "0x"), 16, 64)
 	if err != nil {
-		return err
+		// set latest block to 0 for error
+		mezoLatestBlockGauge.WithLabelValues(moniker, networkID).Set(0)
+		errs = append(errs, fmt.Errorf("invalid latestBlock: %v", err))
+	} else {
+		mezoLatestBlockGauge.WithLabelValues(moniker, networkID).Set(float64(latestBlock))
 	}
-	mezoLatestBlockGauge.WithLabelValues(moniker, networkID).Set(float64(latestBlock))
 
 	ts, err := strconv.ParseUint(strings.TrimPrefix(result.Timestamp, "0x"), 16, 64)
 	if err != nil {
-		return err
+		// set latest timestamp to 0 for error
+		mezoLatestTimestampGauge.WithLabelValues(moniker, networkID).Set(0)
+		errs = append(errs, fmt.Errorf("invalid timestamp: %v", err))
+	} else {
+		mezoLatestTimestampGauge.WithLabelValues(moniker, networkID).Set(float64(ts))
 	}
-	mezoLatestTimestampGauge.WithLabelValues(moniker, networkID).Set(float64(ts))
-
-	return nil
+	return errors.Join(errs...)
 }
 
 func nodeVersion(ctx context.Context, client *rpc.Client, moniker, networkID string) error {
```

Comment:
I think we can use the same as proposed in https://github.com/mezo-org/mezod/pull/450#discussion_r2057588722 for `nodeVersion`.

Reply:
addressed in: d1755782d3115b584b67e6fb763dc3d62386ef81